simulator/encapsulator/InteropEncapsulator.py

```diff
@@ -88,7 +88,7 @@ def handle_timer(self, timer: dict) -> List[_Consequence]:
         self.protocol.handle_timer(timer)
         return self._collect_consequences()
 
-    def handle_packet(self, message: dict) -> List[_Consequence]:
+    def handle_packet(self, message: str) -> List[_Consequence]:

```

Comment:
Please change this function signature in IEncapsulator so that every Encapsulator class has to implement handle_packet using a string message